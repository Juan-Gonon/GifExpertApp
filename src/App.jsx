import { useState } from 'react'
import { AddCategory } from './components/AddCategory'
import { GifGrid } from './components/GifGrid'

function App () {
  const [categories, setCategories] = useState(['One punch'])

  const onAddCategory = ({ inputValue }) => {
    if (categories.includes(inputValue)) return

    setCategories((prevState) => ([...prevState, inputValue]))
  }

  return (
    <>
      <h1>Gif Expert</h1>
      <AddCategory onSetCategory={onAddCategory} />
      {
          categories?.map((element, item) => (
            <GifGrid key={`${element}-${item}`} category={element} />
          ))
      }
    </>
  )
}

export default App
