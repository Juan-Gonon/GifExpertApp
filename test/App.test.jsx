import { describe, it, expect, beforeEach } from 'vitest'
import App from '../src/App'
import { cleanup, render, screen } from '@testing-library/react'

describe('Pruebas en <App />', () => {
  beforeEach(() => {
    cleanup()
  })

  it('Snapshot App', () => {
    const { container } = render(<App />)
    expect(container).toMatchSnapshot()
  })

  it('Render App', () => {
    render(<App />)
    screen.debug()
  })
})
