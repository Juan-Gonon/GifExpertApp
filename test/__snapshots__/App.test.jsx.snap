// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Pruebas en <App /> > Snapshot App 1`] = `
<div>
  <h1>
    Gif Expert
  </h1>
  <form
    action=""
    aria-label="form"
  >
    <input
      placeholder="Buscar gif..."
      type="text"
      value=""
    />
    <button
      class="btn-buscar"
      type="submit"
    >
      Buscar...
    </button>
  </form>
  <h4>
    One punch
  </h4>
  <div
    class="card-grid"
  >
    <div>
      Cargando...
    </div>
  </div>
</div>
`;
